### PR TITLE
bugfix: add a fish shell file to correctly source vars (#3980)

### PR DESCRIPTION
**Pull request by Jiehong**
I don't understand how nix-profile.sh.in gets turned into nix-profile.sh, so this PR might be missing something.

### CODE REVIEW

File: scripts/install-nix-from-closure.sh [hunk 1]
```diff
@@ -247,3 +247,42 @@ variables are set, either log in again, or type
 in your shell.
 EOF
 fi
+
+added=
+p=$HOME/.nix-profile/etc/profile.d/nix.fish
```
Comment (Jiehong) [inline scripts/install-nix-from-closure.sh:252]: I'm assuming `nix-profile.fish.in` is converted to `nix.fish` somehow, but that's unclear.

**Comment by edolstra** [comment]
See #440. The main problem with adding fish support is that it will quickly get out of sync with the bash profile script, since the maintainers (e.g. me) don't use fish.

**Comment by Jiehong** [comment]
> See #440. The main problem with adding fish support is that it will quickly get out of sync with the bash profile script, since the maintainers (e.g. me) don't use fish.

Ah, I didn't see that one. I just read it, and saw what people tried to do there.

I understand your concern here.

Perhaps an alternative would be to display a message if the user has fish installed, saying that nix will only work in bash?

**Comment by cole-h** [comment]
Maybe we could suggest https://github.com/lilyball/nix-env.fish/. When I was using Nix on Arch, that plugin saved my sanity.

**Comment by Jiehong** [comment]
This won't be merged, as the team won't maintain non-bash scripts.

Instead, I made a new PR to display a message instead: https://github.com/NixOS/nix/pull/4033

**Pull request closed without merging**